signer/WebCryptoSigner.js

```diff
@@ -5,33 +5,44 @@ const jsrsasign = require('jsrsasign');
  */
 class WebCryptoSigner {
   /**
-   * @param {String} pkcs1 PKCS#1
+   * @param {String|Object} key key can be a CryptoKey or PKCS#1 PEM
    */
-  constructor(pkcs1) {
-    this.pkcs1 = pkcs1;
+  constructor(key) {
+    if (typeof key == 'string') {
+      this.pkcs1 = key;
+    } else if (typeof key === 'object') {
+      this.key = key;
+    } else {
+      throw new Error(
+          'key type should be either String (PEM) or Object (CryptoKey)'
+      );
+    }
   }
 
   /**
    * @param {Uint8Array} content
    * @return {Uint8Array}
    */
   async sign(content) {
-    if (!this.pkcs8) {
-      this.pkcs8 = this._PKCS1ToPKCS8(this.pkcs1);
+    let key;
+    if (this.key) {
+      key = this.key;
+    } else {
+      if (!this.pkcs8) {
+        this.pkcs8 = await this._PKCS1ToPKCS8(this.pkcs1);
+      }
+      try {
+        key = await this._toCryptoKeyFromPkcs8(this.pkcs8);
```

Comment:
Sorry I realized that it's a bit redundant so maybe it is better.
```suggestion
        key = await this._toCryptoKeyFrom(this.pkcs8);
```